File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.js
function solution(n, left, right) {
    var answer = [];
    for (let i = left; i < right +1; i++) {
        let row = Math.floor(i/n)
        let col = i % n;
        if (row >= col) {
            answer.push(row+1)
        }
        else {
            answer.push(col+1)
        }
    }
    return answer;
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
import java.util.*;

class Solution {
    public int[] solution(int n, long left, long right) {
        List<Long> ans = new ArrayList<>();
        
        for (long idx = left; idx<=right; idx++) {
            long l = (idx / n) + 1;
            long m = (idx % n) + 1;
            ans.add(Math.max(l, m));
        }
        int[] answer = new int[ans.size()];
        for (int i=0; i<ans.size(); i++) {
            answer[i] = ans.get(i).intValue();
        }
        return answer;
    }
}